node/routes/index.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/*
+  Disabling because `status` and `body` are not defined in the `Context` type provided by VTEX.
+  However, they are dynamically added by the middleware in the runtime, and using them works as expected.
+*/
+export async function getEmailRetificationConfig(
+  ctx: Context,
+  next: () => Promise<any>
+) {
+  try {
+    const { storeUserAuthToken } = ctx.vtex
+    const userStoreToken = storeUserAuthToken?.includes('.')
+      ? storeUserAuthToken.split('.')[1]
+      : ''
+
+    const { account } = JSON.parse(
+      Buffer.from(userStoreToken, 'base64').toString()
+    )
+
+    if (account !== ctx.vtex.account) {
+      setForbiddenStatus(ctx)
+
+      return
+    }
+  } catch (e) {
+    setForbiddenStatus(ctx)
+
+    return
+  }
+
+  // @ts-ignore
+  ctx.status = 200
+  // @ts-ignore
+  ctx.body = await ctx.clients.oms.getEmailRetificationConfig()
+
+  await next()
+}
```

Comment:
```suggestion
  try {
    const { storeUserAuthToken } = ctx.vtex
    const { account } = decodeToken(storeUserAuthToken);

    if (account !== ctx.vtex.account) {
      setForbiddenStatus(ctx)
      return
    }
  } catch {
    setForbiddenStatus(ctx)
    return
  }

  // @ts-ignore
  ctx.status = 200
  // @ts-ignore
  ctx.body = await ctx.clients.oms.getEmailRetificationConfig()
  
  await next()
}

function decodeToken(token: string) {
  if (!token || !token.includes('.')) {
    throw new Error('Invalid token');
  }

  const encodedPayload = token.split('.')[1];
  const payload = Buffer.from(encodedPayload, 'base64').toString();
  return JSON.parse(payload);
}
```

Created decodeToken function to extract the responsibility and allow reuse, also added an early return if no token was found

Reply:
Thank you!